reactor-core/src/main/java/reactor/core/publisher/MonoSink.java

```diff
@@ -31,7 +33,45 @@
  *
  * @param <T> the value type emitted
  */
-public interface MonoSink<T> extends ScalarSink<T> {
+public interface MonoSink<T> extends Sink<T> {
```

Comment:
I think this is very problematic. Offering a flux-like API on the `MonoSink`, even if we introduce a specialized API, is bound to be confusing.

ie this API seemingly allows `next(t).next(t).complete()` or even more confusing, `next(t1).next(t2).success(t3)` type of calls.

Reply:
That is changed in the next commit. Sink has orthogonal methods "boolean emitXxx" (complete|error|next). Chaining only brings value to FluxSink if any and it was impossible for processors to be internally sink due to competing error name. Plus it provides for a new feature, returning false if emission failed because the sink is terminated.

Reply:
wouldn't it make implementing the `tryNext` feature harder ? I'm guessing this overlaps a bit, but it could be interesting to be able to distinguish between a `emitNext` call that returns `false` because termination from one that returns `false` because backpressure. wdyt @dfeist ?

Reply:
If we go that route we always bring back the enum `Emission` with various helper `isOk`, `isTerminated`, `isCancelled`, `isDisposed`, `isBackpressured`